website/docs/r/vapp_vm.html.markdown

```diff
@@ -114,6 +114,28 @@ resource "vcd_vapp_vm" "web2" {
   depends_on = ["vcd_vapp.web"]
 }
 
+resource "vcd_vapp_vm" "internalDiskOverride" {
+  vapp_name     = "${vcd_vapp.web.name}"
+  name          = "internalDiskOverride"
+  catalog_name  = "Boxes"
+  template_name = "lampstack-1.10.1-ubuntu-10.04"
+  memory        = 2048
+  cpus          = 2
+  cpu_cores     = 1
+
+  override_template_disk {
+    bus_type         = "paravirtual"
+    size_in_mb       = "22384"
+    bus_number       = 0
+    unit_number      = 0
+    iops             = 0
+    thin_provisioned = true
```

Comment:
With VCD 10 in HTML5 GUI you cannot set thin_provisioned.
And you set the thin_provisioned setting as global ORG-VDC option. You cannot overwrite it per VM.

So thin_provisioned should be a "read" attribute only, not an argument.
Just tested it: Tested true and false, but result is the default ORG-VDC setting:

```hcl
resource "vcd_vapp" "vapp" {
  name = var.hostname
}

resource "vcd_vapp_vm" "instance" {
  name          = var.hostname
  computer_name = var.hostname
  vapp_name     = vcd_vapp.vapp.name
  catalog_name  = var.image_catalog
  template_name = var.image_string
  cpus          = var.cpu
  cpu_cores     = var.core
  memory        = var.memory

  override_template_disk {
    bus_type    = var.bus_type
    size_in_mb  = "40960"
    bus_number  = 0
    unit_number = 0
    thin_provisioned = false
  }

  network {
    type               = "org"
    name               = var.network
    ip_allocation_mode = var.private_ip == null ? "POOL" : "MANUAL"
    ip                 = var.private_ip
    is_primary         = true
  }
}
```

`terraform show`
```
resource "vcd_vapp_vm" "instance" {
    accept_all_eulas               = true
    catalog_name                   = "asys-linux"
    computer_name                  = "test1"
    cpu_cores                      = 1
    cpus                           = 1
    expose_hardware_virtualization = false
    href                           = "https://access.devcloud.arvato-systems.de/api/vApp/vm-6c6bdee2-8c3e-475a-bba2-ad0d8f99a23a"
    id                             = "urn:vcloud:vm:6c6bdee2-8c3e-475a-bba2-ad0d8f99a23a"
    internal_disk                  = [
        {
            bus_number       = 0
            bus_type         = "paravirtual"
            disk_id          = "2000"
            iops             = 0
            size_in_mb       = 40960
            storage_profile  = "Intermediate"
            thin_provisioned = true
            unit_number      = 0
        },
    ]
[...]

    network {
[...]
    }

    override_template_disk {
        bus_number       = 0
        bus_type         = "paravirtual"
        size_in_mb       = 40960
        thin_provisioned = false
        unit_number      = 0
    }
}

```

Reply:
Hi, yes you right. I did test and made changes in internal disk resources. Same changes for overide_template_disk also on the way. Thank you for testing it.

Reply:
done

Reply:
Looks like it was completely removed, but the ask was to keep it as read-only attribute:
> So thin_provisioned should be a "read" attribute only, not an argument.

Reply:
override_template_disk is not read - only input. Thin provisioned value is in read only attribute vm.internal_disk.thin_provisioned

Reply:
Please add that override_template_disk is not read to the "Override template disk" section in the docs before.

Reply:
added

Reply:
Thanks!

Polishing:
`unit_number Changes are ignored on update. This part isn't reread on refresh.`
There's a dot missing at the end of the sentence after "unit_number".

Reply:
done